File: 74-search-a-2d-matrix/search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n=len(matrix)
        m=len(matrix[0])
        lo=0
        hi=(n*m)-1
        while lo<=hi:
            mid=lo+(hi-lo)//2
            row=mid//m
            col=mid%m
            if matrix[row][col]==target:
                return True
            elif target>matrix[row][col]:
                lo=mid+1
            else:
                hi=mid-1
        return False
        

File: 74-search-a-2d-matrix/search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int rows=matrix.length;
        int columns=matrix[0].length;
        int low=0;
        int high=(rows*columns)-1;
        while (low<=high){
            int mid=low+(high-low)/2;
            int col=mid%columns;
            int row=(int) Math.floor((double) mid/columns);
            if(matrix[row][col]==target){
                return true;
            }else if(target>matrix[row][col]){
                low=mid+1;
            }else{
                high=mid-1;
            }
        }

        return false;
        
    }
}